src/oidcendpoint/oidc/token_coop.py

```diff
@@ -196,14 +211,48 @@ def _refresh_token_post_parse_request(self, request, client_id="", **kwargs):
 
         return request
 
+    def _token_exchange_post_parse_request(self, request, client_id="", **kwargs):
```

Comment:
We could override the `parse_request` method, add a `request_cls` param to override `self.request_cls` and provide the proper one depending on the grant_type. This way we can get rid of `*_post_parse_request`

Reply:
I'd rather add a get_request_class() method.